### PR TITLE
S5 display current location on map

### PR DESCRIPTION
**Pull request by koma2323**
- The map can now displays the user's current location and will be centered at that position when opening the Explore tab
- For now the user's current location is refreshed every 5 seconds
- If the user doesn't give access to their location, the camera will be set to some default location 
- Added a button to bring the camera back to the user's current location
- Fixed an issue where clicking on the Explore tab while already in Explore would cause the map to not reload properly
- Added a CircularProgressIndicator while the map is loading

https://github.com/SwEnt-Project-G18/SwEntApp/assets/125990940/f8d15c02-b60c-4cec-a719-d1ddd0a1c965

### CODE REVIEW

**Comment by koma2323** [comment]
some features that were pushed on main were deleted so this is currently broken

**Comment by koma2323** [comment]
there is a commit that says that events are updated but that was removed because there was a bug when an event was removed and the user had the event open

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/Explore.kt
Comment (albertfares) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/Explore.kt:1]: The location logic and camera view seems to be working pretty well. A minor UI issue should be resolved

Comment (albertfares) [reply app/src/main/java/com/github/se/gomeet/ui/mainscreens/Explore.kt:1]: button is now well placed.

**Review by albertfares — APPROVED** [review]
The button position change requested was done.

**Pull request merged**